src/server.cpp

```diff
@@ -2123,6 +2136,61 @@ void Server::sendAddNode(v3s16 p, MapNode n, u16 ignore_id,
 	}
 }
 
+void Server::sendMetadataChanged(const std::list<v3s16> &meta_updates, float far_d_nodes)
+{
+	float maxd = far_d_nodes * BS;
+	NodeMetadataList *meta_updates_list = new NodeMetadataList();
+
+	std::vector<u16> clients = m_clients.getClientIDs();
+	m_clients.lock();
+
+	for (std::vector<u16>::iterator i = clients.begin();
+			i != clients.end(); ++i) {
+		meta_updates_list->clear();
+		Player *player = m_env->getPlayer(*i);
```

Comment:
I have no idea. In the past I had added an errorstream print to the else part of one of these checks, and it complained to me from time to time.

Removing the check here as you propose would possibly lead to the introduction of a hard to reproduce bug.